src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,28 +16,28 @@
 import com.google.cloud.language.v1.LanguageServiceClient;
 import com.google.cloud.language.v1.Sentiment;
 import com.google.gson.Gson;
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /** An item on a todo list. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private final List<Object> commentsList = new ArrayList<>();
   private Key currentTermKey;
-  private final LanguageServiceClient languageService;
+  private LanguageServiceClient languageService;
+  private final DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+  private String schoolName, courseName, profName, termName, termFeedback, professorFeedback;
+  private Long units, termRating, professorRating, workHours, difficulty;
```

Comment:
Do these all need to be global?

Typically best practice is to declare variables as close to where they're used as possible. 

Back in the day declaring at the top was useful for memory estimation/etc, but that's not a concern nowadays and it easy to forget to remove one from being declared if it's not co-located.

Reply:
They don't need to be global, I will fix that now!